core/src/main/java/org/elasticsearch/monitor/jvm/JvmStats.java

```diff
@@ -94,55 +88,84 @@ public static JvmStats jvmStats() {
                  * we just omit the pool in that case!*/
             }
         }
-        stats.mem.pools = pools.toArray(new MemoryPool[pools.size()]);
-
-        stats.threads = new Threads();
-        stats.threads.count = threadMXBean.getThreadCount();
-        stats.threads.peakCount = threadMXBean.getPeakThreadCount();
+        MemoryPool[] memoryPools = pools.toArray(new MemoryPool[pools.size()]);
```

Comment:
I'd be fine with converting these to lists BTW. I'm happy with this as is but I think lists are easier to work with in `StreamInput`?